src/helm/benchmark/static/benchmarking.js

```diff
@@ -399,63 +442,20 @@ $(function () {
         id2originalInstance[instance.id] = instance;
       }
     });
-
+    let instancesHtml = "";
+    let keys = [];
     instances.forEach((instance) => {
       const key = instanceKey(instance);
-      if (key in instanceKeyToDiv) {
-        console.warn(`Two instances with the same key ${key}, skipping`, instance);
-        return;
-      }
-
-      // Assume original version (no perturbation shows up first)
-      if (!instance.perturbation) {
-        $instances.append($('<hr>'));
-      } else {
-        $instances.append($('<br>'));
-      }
-
-      const $instance = $('<div>');
-
-      // For perturbations of an instance, highlight the diff between the unperturbed instance with the same ID
-      const originalInstance = id2originalInstance[instance.id];
-
-      let header;
-      if (!instance.perturbation) {
-        header = `Instance ${instance.id} [split: ${instance.split}]`;
-      } else {
-        header = '...with perturbation: ' + renderPerturbation(instance.perturbation);
-      }
-
-      $instance.append($('<b>').text(header));
-
-      // We can hide the inputs and outputs to focus on the predictions
-      if (!urlParams.hideInputOutput) {
-        $instance.append('<br>');
-        const input = instance.perturbation ? highlightNewWords(instance.input, originalInstance.input) : instance.input;
-
-        // Input
-        $instance.append($('<div>').text('Input:'));
-        $instance.append($('<div>', {class: 'instance-input'}).html(multilineHtml(input)));
-
-        // References
-        if (instance.references.length > 0) {
-          $instance.append($('<div>').text(instance.references.length === 1 ? 'Reference:' : 'References:'));
-          const $references = $('<ul>');
-          instance.references.forEach((reference, referenceIndex) => {
-            const originalReference = instance.perturbation && originalInstance.references[referenceIndex];
-            const output = instance.perturbation ? highlightNewWords(reference.output, originalReference.output) : reference.output;
-            const suffix = reference.tags.length > 0 ? ' ' + ('[' + reference.tags.join(',') + ']').bold() : '';
-            $references.append($('<li>').append([$('<span>', {class: 'instance-reference'}).text(output), suffix]));
-          });
-          $instance.append($references);
-        }
-      }
-      $prediction = $('<div>', {class: 'prediction'});
-      $instance.append($prediction);
-      $instances.append($instance);
-      instanceKeyToDiv[key] = $instance;
+      instancesHtml += instanceTemplate({
+        instance,
+        'unperturbedInstance': id2originalInstance[instance.id],
+        'hideInputOutput': urlParams.hideInputOutput
+      });
+      keys.push(key);
     });
-
+    $instances.html(instancesHtml);
+    const $divs = $instances.find(".instance");
+    keys.forEach((key, index) => {instanceKeyToDiv[key] = $divs.eq(index)});
```

Comment:
Should we double check that `$divs` and `keys`  have the same length?

Reply:
Checked; will print an error if they are not.